### ISSUE TITLE
AMQP Gateway invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by fayak** [argoproj/argo-events#442]
**Describe the bug**

AMQP Gateway doesn't connect to AMQP server and dies without much log, with only a nil pointer dereference
```
$ k -n argo-events logs amqp-gateway-8dddff564-94vdd amqp-events
starting gateway server
INFO[2019-12-17 18:17:46] started processing the event source...        event-source=<redacted>
INFO[2019-12-17 18:17:46] parsing the event source...                   event-source=<redacted>
INFO[2019-12-17 18:17:46] dialing connection...                         event-source=<redacted>
recovered event source <redacted> from error. recover: runtime error: invalid memory address or nil pointer dereference
```

**To Reproduce**

```
apiVersion: argoproj.io/v1alpha1
kind: EventSource
metadata:
  name: amqp-event-source
  namespace: argo-events
spec:
  type: amqp
  amqp:
    <redacted>:
      url: "amqp://<redacted>@rabbitmq:5672"
      URL: "amqp://<redacted>@rabbitmq:5672"
      exchangeName: "<redacted>"
      exchangeType: direct
      routingKey: "<redacted>"
      rateLimit: 5
      ratePeriod: 15
      backoff:
        # nano sec
        duration: 10000000000
        steps: 5
        factor: 2
---
apiVersion: argoproj.io/v1alpha1
kind: Gateway
metadata:
  name: amqp-gateway
  namespace: argo-events
  labels:
    # gateway controller with instanceId "argo-events" will process this gateway
    gateways.argoproj.io/gateway-controller-instanceid: argo-events
spec:
  type: "amqp"
  eventSourceRef:
    name: "amqp-event-source"
  processorPort: "9330"
  eventProtocol:
    type: "HTTP"
    http:
      port: "9300"
  template:
    metadata:
      name: "amqp-gateway"
      labels:
        gateway-name: "amqp-gateway"
    spec:
      containers:
        - name: "gateway-client"
          image: "argoproj/gateway-client:v0.12-rc"
          imagePullPolicy: "Always"
          command: ["/bin/gateway-client"]
        - name: amqp-events
          image: "argoproj/amqp-gateway:v0.12-rc"
          imagePullPolicy: "Always"
          command: ["/bin/amqp-gateway"]
      serviceAccountName: "argo-events-sa"
  service:
    metadata:
      name: amqp-gateway-svc
    spec:
      selector:
        gateway-name: "amqp-gateway"
      ports:
        - port: 1985
          targetPort: 1985
      type: LoadBalancer
  # sensors to send events to
  watchers:
    sensors:
    - name: "moulette-amqp-sensor"
```

**Expected behavior**

Expecting the gateway client to connect to AMQP server and for it to pull events

**Comment by VaibhavPage** [comment]
Taking a look...

**Comment by VaibhavPage** [comment]
The Jitter is missing from `event source`, hence the error. But the gateway should use a default value. I'll submit a patch asap.

**Comment by fayak** [comment]
Even with a jitter is doesn't work, the `backoff` key seems to be wrong as well in the examples, apparently it works with `connectionBackoff`

**Comment by fayak** [comment]
Setting both the `jitter` key and `connectionBackoff` instead of `backoff` seems to be the solution

**Comment by VaibhavPage** [comment]
Ah.. the examples need to be updated.

**Comment by ankitnanglia** [comment]
Is this issue fixed?

I am still seeing the same error with the configuration from the amqp example on github

**Comment by VaibhavPage** [comment]
Try with providing the `connectionBackoff` option. I have put in the fix for v0.12

**Comment by VaibhavPage** [comment]
fixed in v0.12. Let me know if you still face the issue.
